mypy/checker.py

```diff
@@ -3171,6 +3171,9 @@ def named_type(self, name: str) -> Instance:
         # Assume that the name refers to a type.
         sym = self.lookup_qualified(name)
         node = sym.node
+        if isinstance(node, TypeAlias):
+            assert isinstance(node.target, Instance)
```

Comment:
It can't ever be a chain of `TypeAlias`es?

Reply:
It can't, chained aliases are collapsed on sight. An alias can target other types (non-`Instance`) but this function is supposed to be used only with built-in types, that are all instances.